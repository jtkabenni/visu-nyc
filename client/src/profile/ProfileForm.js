import React, { useState, useContext } from "react";
import { useNavigate, Navigate, useRouteError } from "react-router-dom";
import UserContext from "../auth/UserContext";

import NycvisuApi from "../api/api";
/** Form for creating a new snack or drink item.
 *
 * Has state for the name/quantity of the item; on submission,
 * sends {name, qty} to fn rec'd from parent.
 *
 */

const ProfileForm = ({ update }) => {
  const { user } = useContext(UserContext);
  const navigate = useNavigate();
  const [formData, setFormData] = useState({
    firstName: user.firstName,
    lastName: user.lastName,
    email: user.email,
    username: user.username,
    password: "",
  });
  const [formErrors, setFormErrors] = useState([]);

  console.debug(
    "LoginForm",
    "login=",
    typeof login,
    "formData=",
    formData,
    "formErrors",
    formErrors
  );

  /** Send {name, quantity} to parent
   *    & clear form. */

  async function handleSubmit(evt) {
    evt.preventDefault();
    const updateData = {
      firstName: formData.firstName,
      lastName: formData.lastName,
      email: formData.email,
      password: formData.password,
    };
    const updatedUser = await NycvisuApi.updateUser(user.username, updateData);
    console.log(updatedUser);
    update(updatedUser);
    navigate("/");
  }

  /** Update local state w/curr state of input elem */

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };
  if (!user) {
    return <Navigate to="/" />;
  }
  /** render form */

  return (
    <div class="form-container">
      <form onSubmit={handleSubmit}>
        <h3>Update profile</h3>
        <input
          placeholder="Username"
          id="username"
          name="username"
          value={formData.username}
          readonly
        />

        <input
          placeholder="Password"
          type="password"
          id="password"
          name="password"
          value={formData.password}
          onChange={handleChange}
        />

        <input
          placeholder="Email"
          type="email"
          id="email"
          name="email"
          value={formData.email}
          onChange={handleChange}
        />

        <input
          placeholder="First name"
          id="firstName"
          name="firstName"
          value={formData.firstName}
          onChange={handleChange}
        />

        <input
          placeholder="Last name"
          id="lastName"
          name="lastName"
          value={formData.lastName}
          onChange={handleChange}
        />

        <button>Update profile</button>
      </form>
    </div>
  );
};

export default ProfileForm;
