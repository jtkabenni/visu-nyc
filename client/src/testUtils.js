import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  is_admin: true,
  auth_type: "custom",
  maps: [],
};

const demoCurrUserMaps = [];

const UserProvider = ({
  children,
  currUser = demoUser,
  currUserMaps = demoCurrUserMaps,
}) => (
  <UserContext.Provider value={{ currUser, currUserMaps }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
