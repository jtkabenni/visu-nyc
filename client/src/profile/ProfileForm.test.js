import React from "react";
import { render } from "@testing-library/react";
import ProfileForm from "./ProfileForm";
import { UserProvider } from "../testUtils";
import { BrowserRouter as Router } from "react-router-dom";
// TODO: woefully under-tested!

it("matches snapshot", function () {
  const { asFragment } = render(
    <Router>
      <UserProvider>
        <ProfileForm />
      </UserProvider>
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
