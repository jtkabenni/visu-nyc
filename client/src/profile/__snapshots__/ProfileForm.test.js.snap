// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-container"
  >
    <form>
      <h3>
        Update profile
      </h3>
      <input
        id="username"
        name="username"
        placeholder="Username"
        readonly=""
        value="testuser"
      />
      <input
        id="password"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
      <input
        id="email"
        name="email"
        placeholder="Email"
        type="email"
        value="test@test.net"
      />
      <input
        id="firstName"
        name="firstName"
        placeholder="First name"
        value=""
      />
      <input
        id="lastName"
        name="lastName"
        placeholder="Last name"
        value=""
      />
      <button>
        Update profile
      </button>
    </form>
  </div>
</DocumentFragment>
`;
